### DISCUSSION TITLE
Expose registry schemas for external registry builders

### DISCUSSION THREAD

**Discussion by bcorbold** [shadcn-ui/ui#4906] [Ideas]
# Background
Since the latest cli update I've been working on a component registry that leverages the new updates. Currently I've copy/pasted the registry schemas to ensure that our build process generates registry files that work with the new cli, but it would be awesome if there was a way to directly extend from the zod schemas already defined.

Based on todos in the project, it feels like Proposal 1 would be the preferred path. I'd be happy to put together a PR with a first pass if there is interest in the change!

## Proposal 1 - New package
Seems like this idea has [already been on the radar](https://github.com/shadcn-ui/ui/blob/main/packages/shadcn/src/utils/registry/schema.ts#L3). If there is an interest in having this be shared across the monorepo, it would be awesome to publish this as well for external registry builders to leverage as well.

```tsx
import { z } from 'zod'
import { registryItemSchema } from 'shadcn-registry'

// Extend for custom needs while ensuring the structure follows what is expected in the cli
export const myCustomRegistry = registryItemSchema.extend({
  meta: z.object({ foo: z.string() })
})
```

## Proposal 2 - Expose entry in `shadcn` package
Seeing how the schema is already defined in the `shadcn` package, this path could potentially be bundled and exported in the current package. While it would avoid having to expose a new package, it would mean that there is still a dupe entry in `apps/www`

```tsx
import { registryItemSchema } from 'shadcn/registry'
...
```

**Comment by 0xdevalias** [comment]
See also:

- https://github.com/shadcn-ui/ui/discussions/1807
